lib/lev/status.rb

```diff
@@ -2,93 +2,142 @@
 
 module Lev
   class Status
+    attr_reader :id, :status, :progress, :errors
+
     STATE_QUEUED = 'queued'
     STATE_WORKING = 'working'
     STATE_COMPLETED = 'completed'
     STATE_FAILED = 'failed'
     STATE_KILLED = 'killed'
+    STATE_UNKNOWN = 'unknown'
 
     STATES = [
       STATE_QUEUED,
       STATE_WORKING,
       STATE_COMPLETED,
       STATE_FAILED,
-      STATE_KILLED
+      STATE_KILLED,
+      STATE_UNKNOWN
     ].freeze
 
-    attr_reader :uuid
+    def initialize(attrs = {})
+      @id = attrs[:id] || attrs['id'] || SecureRandom.uuid
+      @status = attrs[:status] || attrs['status'] || STATE_QUEUED
+      @progress = attrs[:progress] || attrs['progress'] || set_progress(0)
+      @errors = attrs[:errors] || attrs['errors'] || []
 
-    def initialize(uuid = nil)
-      @uuid = uuid || SecureRandom.uuid
-      save
+      set({ id: id,
+            status: status,
```

Comment:
Can you revert all the changes from `state` to `status`?  I prefer a `state` variable because...

```ruby
status = Lev::Status.find(...)
status.status
```

is confusing to me (better to say `status.state`)